coffee-model/coffee-model-base/src/main/java/hu/icellmobilsoft/coffee/model/base/javatime/listener/TimestampsProvider.java

```diff
@@ -30,90 +32,89 @@
-        for (Property<Object> property : writableResultList) {
-            setProperty(entity, property, sysTime);
+        List<Field> allFields = getAllFields(entity.getClass());
+        for (Field field : allFields) {
+            if (field.isAnnotationPresent(annotationClass)) {
+                setValue(entity, field.getType(), sysTime, field);
+            }
         }
     }
 
-    private List<Property<Object>> getWritableProperties(Object entity, Class<? extends Annotation> annotationClass) {
-        return PropertyQueries.createQuery(entity.getClass())//
-                .addCriteria(new AnnotatedPropertyCriteria(annotationClass))//
-                .getWritableResultList();
-    }
-
-    private void setProperty(Object entity, Property<Object> property, long sysTime) {
-        Object value = sysTime;
+    private void setValue(Object entity, Class<?> fieldClass, long systime, Field field) {
+        Object object = null;
         try {
-            value = now(property.getJavaClass(), sysTime);
-            property.setValue(entity, value);
-        } catch (Exception e) {
-            throw new AuditPropertyException("Failed to write value [" + value + "] to property [" + property.getName() + "] on entity ["
-                    + entity.getClass() + "]: " + e.getLocalizedMessage(), e);
+            if (isCalendarClass(fieldClass)) {
+                Calendar cal = Calendar.getInstance();
+                cal.setTimeInMillis(systime);
+                object = cal;
+            } else if (isDateClass(fieldClass)) {
+                object = fieldClass.getConstructor(Long.TYPE).newInstance(systime);
+            } else if (isOffsetDateTimeClass(fieldClass)) {
+                object = OffsetDateTime.ofInstant(Instant.ofEpochMilli(systime), ZoneId.systemDefault());
+            } else if (isOffsetTimeClass(fieldClass)) {
+                object = OffsetTime.ofInstant(Instant.ofEpochMilli(systime), ZoneId.systemDefault());
+            } else if (isLocalDateTimeClass(fieldClass)) {
+                object = LocalDateTime.ofInstant(Instant.ofEpochMilli(systime), ZoneId.systemDefault());
+            } else if (isLocalDateClass(fieldClass)) {
+                object = LocalDateTime.ofInstant(Instant.ofEpochMilli(systime), ZoneId.systemDefault()).toLocalDate();
+            } else if (isInstantClass(fieldClass)) {
+                object = Instant.ofEpochMilli(systime);
+            } else {
+                throw new IllegalArgumentException("Annotated fieldClass is not a date class: " + fieldClass);
+            }
+            field.setAccessible(true);
+            field.set(entity, object);
+        } catch (IllegalAccessException | NoSuchMethodException | InstantiationException | InvocationTargetException exception) {
+            throw new ProviderException(
+                    "Failed to write value [" + object + "] to entity [" + entity.getClass() + "]: " + exception.getLocalizedMessage(), exception);
```

Comment:
field nevet es osztalyat is tegyuk bele lecci

Reply:
beleirtam